### PR TITLE
[pybind11] Update to 2.7.1

### PR DESCRIPTION
**Pull request by Ryan-rsm-McKenzie**
**Describe the pull request**

- #### What does your PR fix?  
  Updates pybind11 to 2.7.1

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Just an update

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/pybind11/portfile.cmake [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        -DPYBIND11_TEST=OFF
-        -DPYBIND11_FINDPYTHON=ON
-        -DPython3_EXECUTABLE=${PYTHON3}
+        "-DPYBIND11_TEST=OFF"
+        "-DPYBIND11_FINDPYTHON=ON"
```
Comment (JonLiu1993) [inline ports/pybind11/portfile.cmake:15]:
```suggestion
        -DPYBIND11_TEST=OFF
        -DPYBIND11_FINDPYTHON=ON
```

File: ports/pybind11/portfile.cmake [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
     OPTIONS_RELEASE
-        -DPYTHON_IS_DEBUG=OFF
+        "-DPYTHON_IS_DEBUG=OFF"
```
Comment (JonLiu1993) [inline ports/pybind11/portfile.cmake:17]:
```suggestion
        -DPYTHON_IS_DEBUG=OFF
```

File: ports/pybind11/portfile.cmake [hunk 3]
```diff
@@ -1,30 +1,28 @@
...
     OPTIONS_DEBUG
-        -DPYTHON_IS_DEBUG=ON
+        "-DPYTHON_IS_DEBUG=ON"
```
Comment (JonLiu1993) [inline ports/pybind11/portfile.cmake:19]:
```suggestion
        -DPYTHON_IS_DEBUG=ON
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@Ryan-rsm-McKenzie ,Thank for your update,we need to use the new cmake function to update the port in the future, and only need to add "" to the path in the portfile.cmake file.

**Comment by Ryan-rsm-McKenzie** [comment]
What do you mean by the new cmake function?

**Comment by JonLiu1993** [comment]
> What do you mean by the new cmake function?

Like "vcpkg_cmake_install","vcpkg_cmake_configure"

File: ports/pybind11/portfile.cmake [hunk 4]
```diff
@@ -1,30 +1,26 @@
...
-        -DPython3_EXECUTABLE=${PYTHON3}
     OPTIONS_RELEASE
         -DPYTHON_IS_DEBUG=OFF
     OPTIONS_DEBUG
         -DPYTHON_IS_DEBUG=ON
 )
 
-vcpkg_install_cmake()
-vcpkg_fixup_cmake_targets(CONFIG_PATH share/cmake/pybind11)
+vcpkg_cmake_install()
```
Comment (JonLiu1993) [inline ports/pybind11/portfile.cmake:20]:
```suggestion
vcpkg_cmake_install()
vcpkg_fixup_pkgconfig()
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/pybind11/portfile.cmake (hunk 4 above)
Comment (Ryan-rsm-McKenzie) [reply ports/pybind11/portfile.cmake:20]: pybind11 doesn't install a pkgconfig file.

Comment (JonLiu1993) [reply ports/pybind11/portfile.cmake:20]: I accidentally read it wrong

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**